tests/unittests/config/test_cc_ssh.py

```diff
@@ -22,80 +25,92 @@
+
+
+def _replace_options(user: Optional[str] = None) -> str:
+    options = ssh_util.DISABLE_USER_OPTS
+    if user:
+        new_user = user
+    else:
+        new_user = "NONE"
+    options = options.replace("$USER", new_user)
+    options = options.replace("$DISABLE_USER", "root")
+    return options
 
-    def _publish_hostkey_test_setup(self):
-        self.test_hostkeys = {
-            "dsa": ("ssh-dss", "AAAAB3NzaC1kc3MAAACB"),
-            "ecdsa": ("ecdsa-sha2-nistp256", "AAAAE2VjZ"),
-            "ed25519": ("ssh-ed25519", "AAAAC3NzaC1lZDI"),
-            "rsa": ("ssh-rsa", "AAAAB3NzaC1yc2EAAA"),
-        }
-        self.test_hostkey_files = []
-        hostkey_tmpdir = self.tmp_dir()
-        for key_type in cc_ssh.GENERATE_KEY_NAMES:
-            key_data = self.test_hostkeys[key_type]
-            filename = "ssh_host_%s_key.pub" % key_type
-            filepath = os.path.join(hostkey_tmpdir, filename)
-            self.test_hostkey_files.append(filepath)
-            with open(filepath, "w") as f:
-                f.write(" ".join(key_data))
-
-        cc_ssh.KEY_FILE_TPL = os.path.join(hostkey_tmpdir, "ssh_host_%s_key")
-
-    def test_apply_credentials_with_user(self, m_setup_keys):
-        """Apply keys for the given user and root."""
-        keys = ["key1"]
-        user = "clouduser"
-        cc_ssh.apply_credentials(keys, user, False, ssh_util.DISABLE_USER_OPTS)
-        self.assertEqual(
-            [
-                mock.call(set(keys), user),
-                mock.call(set(keys), "root", options=""),
-            ],
-            m_setup_keys.call_args_list,
-        )
-
-    def test_apply_credentials_with_no_user(self, m_setup_keys):
-        """Apply keys for root only."""
-        keys = ["key1"]
-        user = None
-        cc_ssh.apply_credentials(keys, user, False, ssh_util.DISABLE_USER_OPTS)
-        self.assertEqual(
-            [mock.call(set(keys), "root", options="")],
-            m_setup_keys.call_args_list,
-        )
 
-    def test_apply_credentials_with_user_disable_root(self, m_setup_keys):
-        """Apply keys for the given user and disable root ssh."""
-        keys = ["key1"]
-        user = "clouduser"
-        options = ssh_util.DISABLE_USER_OPTS
-        cc_ssh.apply_credentials(keys, user, True, options)
-        options = options.replace("$USER", user)
-        options = options.replace("$DISABLE_USER", "root")
-        self.assertEqual(
-            [
-                mock.call(set(keys), user),
-                mock.call(set(keys), "root", options=options),
-            ],
-            m_setup_keys.call_args_list,
-        )
+@mock.patch(MODPATH + "ssh_util.setup_user_keys")
+class TestHandleSsh:
+    """Test cc_ssh handling of ssh config."""
 
-    def test_apply_credentials_with_no_user_disable_root(self, m_setup_keys):
-        """Apply keys no user and disable root ssh."""
-        keys = ["key1"]
-        user = None
+    @pytest.mark.parametrize(
+        "args,kwargs",
```

Comment:
Why `kwargs`? It's an empty dict in all of these definitions, so we shouldn't need it at all.